integration-tests/contract-permits.spec.ts

```diff
@@ -0,0 +1,242 @@
+          code: `{ parameter nat;
+            storage address;
+            code { DUP;
+                    CAR;
+                    DIP { CDR };
+                    PUSH nat 42;
+                    COMPARE;
+                    EQ;
+                    IF {  }
+                      { PUSH string "not 42";
+                        FAILWITH };
+                    DUP;
+                    SENDER;
+                    COMPARE;
+                    EQ;
+                    IF {  }
+                      { PUSH string "not admin";
+                        FAILWITH };
+                    NIL operation;
+                    PAIR }; }`,
+          storage: localTez1Pkh
+      });
+
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      const contract = await op.contract();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+      //submit the admin parameter to the contract
+      Tezos.contract
+      .at(contract.address)
+      .then((c) => {
+      return c.methods.permit("admin").send();
+      })
+      .then(async (op) => {
+        await op.confirmation(3);
+        return op.hash;
+      })
+      .catch(Error) ;
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      done();
+
+      //submit the 42 parameter to the contract
+      Tezos.contract
+      .at(contract.address)
+      .then((c) => {
+      return c.methods.permit("42").send();
+      })
+      .then(async (op) => {
+        await op.confirmation(3);
+        return op.hash;
+      })
+      .catch(Error) ;
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      done();
+    });
+  });
+
+  test('Permit can be submitted and used', async (done) => {
+
+      const op = await Tezos.contract.originate({
+        code: permit_admin_42,
+        storage:
+        {
+        0: new MichelsonMap(),
+        1: 0,
+        2: 'tz1bDCu64RmcpWahdn9bWrDMi6cu7mXZynHm'
+        },
+        });
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(op.status).toEqual('applied')
+
+        const permit_parameter = 'Pair "edpkuPTVBFtbYd6gZWryXypSYYq6g7FvyucwphoU78T1vmGkbhj6qb" (Pair "edsigtfkWys7vyeQy1PnHcBuac1dgj2aJ8Jv3fvoDE5XRtxTMRgJBwVgMTzvhAzBQyjH48ux9KE8jRZBSk4Rv2bfphsfpKP3ggM" 0x0f0db0ce6f057a8835adb6a2c617fd8a136b8028fac90aab7b4766def688ea0c)'
+
+        //submit the permit_parameter to the contract
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.permit(permit_parameter).send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+          .catch(Error) ;
+          expect(op.hash).toBeDefined();
```

Comment:
The lines 150 and 151 are testing the constant op created at line 122 instead of the operation representing the call of the permit method

Reply:
I removed the lines